src/python/pants/backend/python/goals/package_pex_binary.py

```diff
@@ -141,7 +143,7 @@ async def package_pex_binary(
     request = PexFromTargetsRequest(
         addresses=[field_set.address],
         internal_only=False,
-        main=resolved_entry_point.val or field_set.script.value,
+        main=resolved_entry_point.val or field_set.script.value or field_set.executable.value,
```

Comment:
Re my question on precedence, does this always get an entry point? Can its `.val` be None? How is this case different than the one below?

Reply:
In this case, yes, as this is the general purpose case and can be anything.

In the other file, it is a utility function only used for the run goal, so it always has an entry point.